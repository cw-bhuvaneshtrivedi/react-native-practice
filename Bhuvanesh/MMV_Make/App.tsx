import { StatusBar } from "expo-status-bar";
import Home from "./components/Home";
import { useState } from "react";
import { Button, Platform, StyleSheet, View } from "react-native";

export default function App() {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <>
      <StatusBar style="auto" />
      <View style={styles.body}>
        {isOpen && <Home setOpen={setIsOpen} />}
        <Button title="Open" onPress={() => setIsOpen(true)} testID="Open" />
      </View>
    </>
  );
}

const styles = StyleSheet.create({
  body: {
    backgroundColor: "black",
    paddingTop: Platform.OS === "android" ? 24 : 0,
    // justifyContent: "center",
  },
});
