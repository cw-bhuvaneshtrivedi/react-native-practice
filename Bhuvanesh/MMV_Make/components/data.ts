const data = [
  [
    "Item aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "Item aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "Item aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  ],
  ["Item 1 1", "Item 1 2", "Item 1 3"],
];

export default data;
