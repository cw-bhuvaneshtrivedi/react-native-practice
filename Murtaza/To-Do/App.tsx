import React, { useState } from "react";
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  FlatList,
  StatusBar,
  Platform
} from "react-native";

type Todo = {
  id: string;
  text: string;
};

const App = () => {
  const [task, setTask] = useState<string>("");
  const [todos, setTodos] = useState<Todo[]>([]);

  const addTodo = () => {
    if (task.trim() === "") return;

    const newTodo: Todo = {
      id: new Date().getTime().toString(),
      text: task,
    };

    setTodos([...todos, newTodo]);
    setTask("");
  };

  const removeTodo = (id: string) => {
    const updatedTodos = todos.filter((todo) => todo.id !== id);
    setTodos(updatedTodos);
  };

  const renderTodoItem = ({ item }: { item: Todo }) => (
    <View style={styles.todoItem}>
      <Text>{item.text}</Text>
      <TouchableOpacity onPress={() => removeTodo(item.id)}>
        <Text style={styles.deleteButton}>Delete</Text>
      </TouchableOpacity>
    </View>
  );

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Todo App</Text>
      <TextInput
        style={styles.input}
        placeholder="Enter your task"
        value={task}
        onChangeText={(e) => setTask(e)}
      />
      <TouchableOpacity style={styles.addButton} onPress={addTodo}>
        <Text style={styles.addButtonText}>Add Task</Text>
      </TouchableOpacity>
      <FlatList
        data={todos}
        renderItem={renderTodoItem}
        keyExtractor={(item) => item.id}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: "#fff",
    paddingTop: Platform.OS === "android" ?  StatusBar.currentHeight + 10 : 0,
  },
  title: {
    fontSize: 24,
    textAlign: "center",
    fontWeight: "bold",
    marginBottom: 20,
  },
  input: {
    height: 40,
    borderColor: "gray",
    borderWidth: 1,
    borderRadius: 5,
    marginBottom: 10,
    paddingLeft: 10,
  },
  addButton: {
    backgroundColor: "blue",
    padding: 10,
    borderRadius: 5,
    alignItems: "center",
    marginBottom: 10,
  },
  addButtonText: {
    color: "#fff",
    fontSize: 16,
    fontWeight: "bold",
  },
  todoItem: {
    flexDirection: "row",
    justifyContent: "space-between",
    backgroundColor: "#eee",
    padding: 10,
    borderRadius: 5,
    marginBottom: 10,
  },
  deleteButton: {
    color: "red",
  },
});

export default App;
