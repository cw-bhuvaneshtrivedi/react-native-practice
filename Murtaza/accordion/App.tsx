import {
  Platform,
  StatusBar,
  StyleSheet,
  Dimensions,
  Text,
  View,
  TouchableOpacity,
} from "react-native";
import React, { useState } from "react";

import SearchPage from "./src/components/SearchPage";
export default function App() {
  const [click, setClick] = useState(false);
  const handlePress = () => {
    setClick(true);
  };
  return (
    <View style={styles.container}>
      {click ? <SearchPage clicked={setClick} /> : null}
      <View>
        <TouchableOpacity onPress={handlePress} style={styles.page}>
          <View style={styles.button}>
            <Text style={styles.buttonText}>Click Me!</Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#efefef",
    // alignItems: 'center',
    // justifyContent: 'center',
    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight + 10 : 0,
  },
  page: { 
    justifyContent: "center", 
    height: Dimensions.get("screen").height },
  button: {
    // flex: 1,
    backgroundColor: "blue",
    padding: 10,
    borderRadius: 5,
    // alignItems: "center",
    justifyContent: "center",
    alignSelf: "center",
  },
  buttonText: {
    color: "white",
    fontSize: 25,
  },
});
