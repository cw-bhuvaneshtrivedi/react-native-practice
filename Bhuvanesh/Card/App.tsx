import React from "react";
import { StatusBar } from "expo-status-bar";
import { View } from "react-native";
import globalStyles from "./Styles";
import { Card } from "./components/Card";
import { FlashList } from "@shopify/flash-list";
import tmp from "./tmp.json";

export default function App() {
  return (
    <React.StrictMode>
      <View style={globalStyles.body}>
        <StatusBar />
        <FlashList
          data={tmp.models}
          estimatedItemSize={360}
          renderItem={({ item }) => <Card carData={item} />}
        />
        {/* <Card /> */}
      </View>
    </React.StrictMode>
  );
}
