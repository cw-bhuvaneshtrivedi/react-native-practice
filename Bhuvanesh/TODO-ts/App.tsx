import { useState } from "react";
import { Button, FlatList, Text, TextInput, View } from "react-native";

import { Todo } from "./components/todo";
import styles from "./Styles";
import colors from "./constants/colors";

export default function App() {
  const [input, setInput] = useState("");
  const [tasks, setTasks] = useState([
    { id: "1", task: "Create a TODO App" },
    { id: "2", task: "View YT video" },
  ]);

  return (
    <View style={styles.body}>
      <Text style={styles.title}>My Todos</Text>
      <View style={styles.rowFlex}>
        <TextInput
          value={input}
          style={styles.input}
          onChangeText={setInput}
          placeholder="Add Something to do ..."
        />
        <Button
          color={colors.blue}
          title="Save"
          onPress={() => {
            if (input == "") {
              alert("Cannot add empty todo");
              return;
            }
            setTasks((tasks) => [
              ...tasks,
              { id: Date.now().toString(), task: input },
            ]);
            setInput("");
          }}
        />
      </View>
      {tasks.length > 0 ? (
        <FlatList
          data={tasks}
          keyExtractor={(item) => item.id}
          renderItem={({ item }) => (
            <Todo
              id={item.id}
              todo={item.task}
              deleteTodo={() => {
                setTasks(tasks.filter((task) => task.id !== item.id));
                alert("Todo deleted");
              }}
            />
          )}
        />
      ) : (
        <Text>No todos left type to add</Text>
      )}
    </View>
  );
}
