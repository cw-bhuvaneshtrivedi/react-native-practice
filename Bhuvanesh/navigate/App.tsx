import { Provider } from "react-redux";
import { NavigationContainer } from "@react-navigation/native";

import store from "./store";
import { StatusBar } from "react-native";
import Stack from "./routes/homeStack";
import { DrawerContainer } from "./routes/DrawerContainer";
import { Stack2 } from "./components/Stack2";
import { Stack1 } from "./components/Stack1";
import { Stack3 } from "./components/Stack3";

export default function App() {
  return (
    <Provider store={store}>
      <StatusBar />
      <NavigationContainer>
        <Stack.Navigator
          screenOptions={{
            headerTintColor: "white",
            headerStyle: { backgroundColor: "tomato" },
          }}
        >
          <Stack.Screen
            options={{ headerShown: false }}
            name="Drawer"
            component={DrawerContainer}
          />
          <Stack.Screen name="Stack1" component={Stack1} />
          <Stack.Screen name="Stack2" component={Stack2} />
          <Stack.Screen name="Stack3" component={Stack3} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
}
