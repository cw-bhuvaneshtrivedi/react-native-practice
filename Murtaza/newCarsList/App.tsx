import React from "react";
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  FlatList,
  SafeAreaView,
  StatusBar,
  Platform,
  Dimensions,
} from "react-native";
import { FlashList } from "@shopify/flash-list";
import {Card} from "./src/components/Card";
import {data} from "./src/data/Data";
export default function App() {
  return (
    <SafeAreaView style={styles.container}>
        {/* <Text>check...</Text> */}
      <FlashList
        data={data}
        renderItem={({ item }) => <Card item={item} />}
        estimatedItemSize={284}
        getItemType={(item) => item.modelId}
      />
      
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    // flex: 1,
    backgroundColor: '#fff',
    height: Dimensions.get( 'window' ).height,
    width: Dimensions.get('window').width,
    paddingTop: Platform.OS === "android" ?  StatusBar.currentHeight : 0,
  },
});
