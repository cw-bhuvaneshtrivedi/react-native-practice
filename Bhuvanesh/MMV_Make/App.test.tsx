import { fireEvent, render, waitFor } from "@testing-library/react-native";
import App from "./App";
import data from "./components/data";

describe("App Component", () => {
  it("Not showing Home component", () => {
    const app = render(<App />);
    expect(app).toMatchSnapshot("App_1.test.tsx.snap");
  });

  it("should show home component", async () => {
    // jest
    //   .spyOn(window, "fetch")
    //   .mockImplementationOnce(() => Promise.resolve(data));
    global.fetch = jest.fn((url: string) =>
      Promise.resolve({ status: 200, json: () => Promise.resolve(data) })
    ) as jest.Mock;
    const app = render(<App />);
    const button = app.getByTestId("Open");
    await waitFor(() => {
      fireEvent.press(button);
      expect(app).toMatchSnapshot("App_2.test.tsx.snap");
    });
  });
});
