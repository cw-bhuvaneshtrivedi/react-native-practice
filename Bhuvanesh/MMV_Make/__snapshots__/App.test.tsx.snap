// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Not showing Home component: App_1.test.tsx.snap 1`] = `
<View
  style={
    {
      "backgroundColor": "black",
      "paddingTop": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="Open"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Open
      </Text>
    </View>
  </View>
</View>
`;

exports[`App Component should show home component: App_2.test.tsx.snap 1`] = `
<View
  style={
    {
      "backgroundColor": "black",
      "paddingTop": 0,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "translateY": 1334,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "backgroundColor": "white",
        "height": 1334,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#484848",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Select Your Make or Model
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <TextInput
      placeholder="Type to Select Make"
      style={
        {
          "borderColor": "#d5d5d5",
          "borderWidth": 1,
          "height": 44,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 31,
          "paddingLeft": 16,
        }
      }
    />
    <ActivityIndicator
      size="large"
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="Open"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Open
      </Text>
    </View>
  </View>
</View>
`;
